trio/tests/test_highlevel_open_unix_stream.py

```diff
@@ -0,0 +1,35 @@
+import socket
+import tempfile
+
+import pytest
+
+from trio import open_unix_socket, Path
+
+try:
+    from socket import AF_UNIX
+except ImportError:
+    pytestmark = pytest.mark.skip("Needs unix socket support")
+
+
+async def test_open_unix_socket():
+    name = Path() / tempfile.gettempdir() / "test.sock"
+    try:
+        await name.unlink()
+    except OSError:
+        pass
+
+    # server socket
+    serv_sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+    serv_sock.bind(name.__fspath__())
+    serv_sock.listen(1)
+
+    # for some reason the server socket can accept the connection AFTER it has
+    # been opened, so use that to test here
+    unix_socket = await open_unix_socket(name.__fspath__())
```

Comment:
Can you also test with passing the `Path` object directly to `open_unix_socket`? (On 3.5 you can use a `trio.Path` to make sure it has an `__fspath__` method.)

Reply:
Will do later today.